### PR TITLE
feat: add support for Hive partitioning options when creating external tables

### LINKED ISSUE #235: Need support for Hive partitioning options when creating external table
The SDK doesn't support setting Hive partitioning options when creating an external table in BigQuery. Here is the corresponding API doc: https://cloud.google.com/bigquery/docs/reference/rest/v2/tables#HivePartitioningOptions.

### PR DESCRIPTION
**Pull request by ekanth**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #235 ☕️

### CODE REVIEW

**Comment by ekanth** [comment]
@googlebot I signed it!

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java [hunk 1]
```diff
@@ -212,6 +215,13 @@ public Boolean ignoreUnknownValues() {
   @Nullable
   public abstract Boolean getAutodetect();
 
+  /**
+   * [Experimental] Returns the HivePartitioningOptions when the data layout follows Hive partitioning convention
+   */
+  @SuppressWarnings("unchecked")
+  @Nullable
+  public abstract HivePartitioningOptions getHivePartitioningOptions();
```
Comment (stephaniewang526) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: this breaks binary compatibility

**Comment by stephaniewang526** [comment]
In general, when you make a commit please be sure to:

1) make initial commit message in the format - "feat: blah blah..." (note that "blah" has a space in front of the colon and is in lowercase)

2) run maven coveo code formatter so that code format is correct: `mvn com.coveo:fmt-maven-plugin:format`

3) check and make sure all builds succeed and correct any build failures. In this case, abstract classes were added which break binary compatibility -- 
```
[ERROR] 7013: com.google.cloud.bigquery.ExternalTableDefinition: Abstract method 'public com.google.cloud.bigquery.HivePartitioningOptions getHivePartitioningOptions()' has been added
[ERROR] 7013: com.google.cloud.bigquery.ExternalTableDefinition$Builder: Abstract method 'public com.google.cloud.bigquery.ExternalTableDefinition$Builder setHivePartitioningOptions(com.google.cloud.bigquery.HivePartitioningOptions)' has been added
```
we don't want to do this because this renders our client unstable and would break customer's existing implementation. I suggest to try either implement without abstract and add default implementation or encapsulate with an inner method.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java (hunk 1 above)
Comment (ekanth) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: Thanks @stephaniewang526 for the review. I will update the patch and send again.

Comment (ekanth) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: @stephaniewang526, @pmakani I updated the patch and I wonder if the tests would automatically run?

Comment (ekanth) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: @stephaniewang526, thanks for triggering the tests. It was my bad with the previous patch - I missed a line that I fixed locally related to the test failures. Could you please trigger the tests again?

Comment (ekanth) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: Thanks @stephaniewang526. All tests passed now.

**Pull request merged**